lib/jekyll/excerpt.rb

```diff
@@ -50,6 +50,13 @@ def path
     def relative_path
       @relative_path ||= File.join(doc.relative_path, "#excerpt")
     end
+    
+    # The base filename of the excerpt.
+    #
+    # Returns the base filename for the doc this excerpt belongs.
+    def basename
+      @basename ||= doc.basename
+    end
```

Comment:
This is all that needs to be implemented.
No need of additional comments
```suggestion
    def name
      @name ||= @obj.doc.basename
    end
```

Reply:
I changed according to your suggestion.